src/librustc/middle/def.rs

```diff
@@ -29,8 +29,9 @@ pub enum Def {
     DefAssociatedConst(DefId),
     DefLocal(DefId, // def id of variable
              ast::NodeId), // node id of variable
-    DefVariant(DefId /* enum */, DefId /* variant */, bool /* is_structure */),
-    DefTy(DefId, bool /* is_enum */),
+    DefVariant(DefId /* enum */, DefId /* variant */),
```

Comment:
Yeah, and rename `DefVariant` to `Def::Variant` etc.
It would be a huge code churn and I'd like to separate semantic changes done in this PR from pure renaming. But I can do it in a separate commits if reviewer agrees.